src/components/auth/ResetPasswordForm.jsx

```diff
@@ -0,0 +1,86 @@
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
+import FormInput from "../common/FormInput";
+import { useNavigate } from "react-router-dom";
+
+const ResetPasswordForm = () => {
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [message, setMessage] = useState('');
+  const navigate = useNavigate();
+  const [token, setToken] = useState('');
+
+  useEffect(() => {
+    const urlToken = new URLSearchParams(window.location.search).get('token');
+    console.log('Extracted token from URL:', urlToken);
+    if (urlToken) {
+      setToken(urlToken);
+    } else {
+      console.error('No token found in URL');
+    }
+  }, []);
+
+  const handleSubmit = async (e) => {
```

Comment:
This function can also benefit from some minor refactoring for clarity. It handles password comparison and a request to change the password.

Reply:
Fix it